### PR TITLE
#27: o.l.c.m.NotEmpty matcher

### PR DESCRIPTION
**Pull request by dgroup**
#27

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#31](https://github.com/llorllale/cactoos-matchers/issues/31) is now in scope, role is `REV`

**Comment by dgroup** [comment]
https://travis-ci.org/llorllale/cactoos-matchers/builds/457480519?utm_source=github_status&utm_medium=notification
Oracle JDK8 (java-8-oracle)
```java
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.168 sec <<< FAILURE! - in org.llorllale.cactoos.matchers.RunsInThreadsTest
matchNegative(org.llorllale.cactoos.matchers.RunsInThreadsTest)  Time elapsed: 0.167 sec  <<< FAILURE!
java.lang.AssertionError: 
Can concurrently modify HashMap
Expected: <false>
     but: was <true>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.llorllale.cactoos.matchers.RunsInThreadsTest.matchNegative(RunsInThreadsTest.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:393)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

File: src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * The matcher to check that text is not empty.
+ *
+ * @since 1.0.0
+ * @checkstyle ProtectedMethodInFinalClassCheck (100 lines)
+ */
+public final class NotEmpty extends TypeSafeDiagnosingMatcher<String> {
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java]: @dgroup regarding the name - see https://github.com/llorllale/cactoos-matchers/issues/27#issuecomment-440482652

Comment (dgroup) [reply src/main/java/org/llorllale/cactoos/matchers/NotEmpty.java]: Agree, this name is better.

File: src/main/java/org/llorllale/cactoos/matchers/NotBlank.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public void describeTo(final Description desc) {
+        desc.appendText("[blank]");
+    }
+
+    @Override
+    protected boolean matchesSafely(final String text,
+        final Description desc) {
+        desc.appendText("The string ")
+            .appendValue(text)
+            .appendText(" is not empty");
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/NotBlank.java]: @dgroup can we change this text now that this is called `NotBlank`?

**Comment by dgroup** [comment]
@llorllale , is there any review comments to fix?

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:llorllale/cactoos-matchers#31` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#31](https://github.com/llorllale/cactoos-matchers/issues/31) is now out of scope

**Pull request merged**